streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -611,65 +628,170 @@ public StateStore getStateStore(final String name) {
     }
 
     /**
-     * Get the {@link KeyValueStore} with the given name.
+     * Get the {@link KeyValueStore} or {@link TimestampedKeyValueStore} with the given name.
      * The store can be a "regular" or global store.
      * <p>
+     * If the registered store is a {@link TimestampedKeyValueStore} this method will return a value-only query
+     * interface. <strong>It is highly recommended to update the code for this case to avoid bugs and to use
+     * {@link #getTimestampedKeyValueStore(String)} for full store access instead.</strong>
+     * <p>
      * This is often useful in test cases to pre-populate the store before the test case instructs the topology to
      * {@link #pipeInput(ConsumerRecord) process an input message}, and/or to check the store afterward.
      *
      * @param name the name of the store
-     * @return the key value store, or {@code null} if no {@link KeyValueStore} has been registered with the given name
+     * @return the key value store, or {@code null} if no {@link KeyValueStore} or {@link TimestampedKeyValueStore}
+     * has been registered with the given name
      * @see #getAllStateStores()
      * @see #getStateStore(String)
+     * @see #getTimestampedKeyValueStore(String)
      * @see #getWindowStore(String)
+     * @see #getTimestampedWindowStore(String)
      * @see #getSessionStore(String)
+     * @see #getTimestampedSessionStore(String)
      */
     @SuppressWarnings({"unchecked", "WeakerAccess"})
     public <K, V> KeyValueStore<K, V> getKeyValueStore(final String name) {
         final StateStore store = getStateStore(name);
+        if (store instanceof TimestampedKeyValueStore) {
+            log.warn("Method #getTimestampedKeyValueStore() should be used to access a TimestampedKeyValueStore.");
```

Comment:
I'm not sure I follow the need for this log message. As I read it, the method does exactly what it claims to do. But maybe I'm just not being imaginative.
Can you explain why you think folks will be confused by the method's behavior?

Reply:
I don't think people would be confused. The idea is, to educate/remind DSL users to update their code, because we do the store switch "under the hood". Not sure if WARN is the right log level (or if we want to log this at all). It was an idea, do I added it for discussion. Thoughts?

Reply:
Btw: My plan was to add the same log message to IQ code (in a follow up PR that adds IQ -- in this PR, we only add the interfaces and store-types, but IQ is not fully implemented in this PR.)

Reply:
Instead of `WARN` maybe `INFO` as it's not "harmful"  but a subtle reminder for DSL users to update.